etna/search/views.py

```diff
@@ -313,6 +314,18 @@ def get_context_data(self, **kwargs: Any) -> Dict[str, Any]:
         )
         return super().get_context_data(**kwargs)
 
+    def set_session_info(self) -> None:
+        """
+        Usually called where there are links to the record details page in the search results.
+        Sets session in order for it to be used in record details page when navigating from search results.
+        """
+        back_to_search_url_timestamp = timezone.now()
+        self.request.session[
+            "back_to_search_url_timestamp"
+        ] = back_to_search_url_timestamp.strftime("%Y%m%d-%H%M%S")
+        self.request.session["back_to_search_url"] = self.request.get_full_path()
```

Comment:
Tiny tiny thing: I would set the URL first, as the timestamp means nothing with it.

Reply:
Done done :)